python/doc/examples/reliability_sensitivity/reliability/plot_nais.py

```diff
@@ -0,0 +1,248 @@
+
+# %%
+g = ot.MemoizeFunction(g)
+
+# %%
+# Create the output random vector :math:`Y = g(X)`:
+
+# %%
+Y = ot.CompositeRandomVector(g, X)
+
+# %%
+# Create the event :math:`\{ Y = g(X) \leq 0 \}`
+# ----------------------------------------------
+
+# %%
+threshold = 0.0
+myEvent = ot.ThresholdEvent(Y, ot.Less(), threshold)
+
+# %%
+# Evaluate the probability with the NAIS technique
+# -----------------------------------------------------------
+
+# %%
+rhoQuantile = 0.1
+algo = ot.NAIS(myEvent,rhoQuantile)
+
+# %%
+# Now you can run the algorithm
+
+# %%
+algo.run()
+
+# %%
+result = algo.getResult()
+proba = result.getProbabilityEstimate()
+print('Proba NAIS = ',  proba)
+print('Current coefficient of variation = ',
+      result.getCoefficientOfVariation())
+
+# %%
+# The length of the confidence interval of level :math:`95\%` is:
+
+# %%
+length95 = result.getConfidenceLength()
+print('Confidence length (0.95) = ', result.getConfidenceLength())
+
+# %%
+# which enables to build the confidence interval:
+
+# %%
+print('Confidence interval (0.95) = [', proba -
+      length95/2, ', ', proba + length95/2, ']')
+
+# %%
+# Draw the NAIS samples used by the algorithm
+# ---------------------------------------------
+#
+# The following manipulations are possible only if you have created a *MemoizeFunction* that enables to store all the inputs and outputs of the function :math:`g`.
+#
+# Get all the inputs and outputs where :math:`g` were evaluated:
+
+# %%
+inputNAIS = g.getInputHistory()
+outputNAIS = g.getOutputHistory()
+nTotal = inputNAIS.getSize()
+print('Number of evaluations of g = ', nTotal)
+
+# %%
+# Within each step of the algorithm, a sample of size :math:`N` is created, where:
+
+# %%
+N = algo.getMaximumOuterSampling()*algo.getBlockSize()
+print('Size of each subset = ', N)
+
+# %%
+# You can get the number :math:`Ns` of steps with:
+
+# %%
+Ns =  int(nTotal / N)
+print('Number of steps = ', Ns)
+
+# %%
+# Now, we can split the initial sample into NAIS samples of size :math:`Ns`:
+
+# %%
+listNAISSamples = list()
+listOutputNAISSamples = list()
+for i in range(Ns):
+    listNAISSamples.append(inputNAIS[i*N:i*N + N])
+    listOutputNAISSamples.append(outputNAIS[i*N:i*N + N]) 
+
+# %%
+# And get all the levels defining the intermediate and final thresholds given by the empirical quantiles of each NAIS output sample:
+
+# %%
+levels = ot.Point()
+for i in range(Ns-1):
+    levels.add(listOutputNAISSamples[i].computeQuantile(rhoQuantile)[0])
+levels.add(threshold)
+
+# %%
+# The following graph draws each NAIS sample and the frontier :math:`g(x_1, x_2) = l_i` where :math:`l_i` is the threshold at the step :math:`i`:
+
+# %%
+graph = ot.Graph()
+graph.setAxes(True)
+graph.setGrid(True)
+graph.setTitle('NAIS samples')
+graph.setXTitle(r'$x_1$')
+graph.setYTitle(r'$x_2$')
+graph.setLegendPosition('bottomleft')
```

Comment:
You can also use this Graph ctor to pass most arguments:
`Graph(title, xTitle, yTitle, showAxes, legendPosition)`